src/main/models/user.py

```diff
@@ -1,40 +1,84 @@
-
-    __tablename__ = "users"
-    id = Column(String, primary_key=True)
-    email = Column(String, unique=True, nullable=False)
-    full_name = Column(String, nullable=False)
-
-    @classmethod
-    async def create(cls, db: AsyncSession, **kwargs: dict[str, Any]):
-        """Create a new user."""
-        transaction = cls(id=uuid4().hex, **kwargs)
-        db.add(transaction)
-        await db.commit()
-        await db.refresh(transaction)
-        return transaction
-
-    @classmethod
-    async def get(cls, db: AsyncSession, id: str):
-        """Get user by ID."""
-        try:
-            transaction = await db.get(cls, id)
-        except NoResultFound:
-            return None
-        return transaction
-
-    @classmethod
-    async def get_all(cls, db: AsyncSession):
-        """Fetch all users."""
-        return (await db.execute(select(cls))).scalars().all()
+class OAuthAccount(SQLAlchemyBaseOAuthAccountTableUUID, Base):
+    """Google Oauth specific information for a user."""
+
+    pass
+
+
+class User(SQLAlchemyBaseUserTableUUID, Base):
+    """Base user class, extending FastAPI-users."""
+
+    oauth_accounts: Mapped[list[OAuthAccount]] = relationship(
+        "OAuthAccount", lazy="joined"
+    )
+
+
+UserDB = SQLAlchemyUserDatabase[User, uuid.UUID]
+
+
+async def get_user_db(session: AsyncSession = Depends(get_db)):
+    """Return the FastAPI-users database object."""
+    db: UserDB = SQLAlchemyUserDatabase(session, User, OAuthAccount)
+    yield db
+
+
+class UserManager(UUIDIDMixin, BaseUserManager[User, uuid.UUID]):
+    """Define handlers for user events."""
+
+    reset_password_token_secret = get_secret("RESET_PASSWORD_TOKEN_SECRET")
+    verification_token_secret = get_secret("VERIFICATION_TOKEN_SECRET")
+
+    async def on_after_register(self, user: User, request: Request | None = None):
+        """Handle user registation event."""
+        print(f"User {user.id} has registered.")
+
+    async def on_after_forgot_password(
+        self, user: User, token: str, request: Request | None = None
+    ):
+        """Handle user forgot password event."""
+        print(f"User {user.id} has forgot their password. Reset token: {token}")
+
+    async def on_after_request_verify(
+        self, user: User, token: str, request: Request | None = None
+    ):
+        """Handle user verification event."""
+        print(f"Verification requested for user {user.id}. Verification token: {token}")
+
+
+async def get_user_manager(user_db: UserDB = Depends(get_user_db)):
+    """Return an instance of the UserManager class."""
+    yield UserManager(user_db)
+
+
+class UserRead(schemas.BaseUser[uuid.UUID]):
+    """Pydantic schema for a user object on read."""
+
+    pass
+
+
+class UserCreate(schemas.BaseUserCreate):
+    """Pydantic schema for a user object on create."""
+
+    pass
+
+
+class UserUpdate(schemas.BaseUserUpdate):
+    """Pydantic schema for a user object on update."""
+
+    pass
```

Comment:
Should we add TODOs here to come back to these?

Reply:
I don't think we have any changed to make here just yet, this mostly buys us flexibility in case we track more info on users
https://fastapi-users.github.io/fastapi-users/11.0/configuration/schemas/